bestie/server/test_result.go

```diff
@@ -105,58 +139,50 @@ func handleTestResultEvent(bazelBuildEvent bes.BuildEvent, streamId *build.Strea
 	sbuf.WriteString(fmt.Sprintf("\tinvocationSha: %s\n", stream.invocationSha))
 	debugPrintln(sbuf.String())
 
-	var outFileName, outFileUri string
+	var errs []error
+	var fileName, fileUri string
 	for _, of := range m.GetTestActionOutput() {
-		if strings.HasSuffix(of.GetName(), "outputs.zip") {
-			outFileName = of.GetName()
-			outFileUri = of.GetUri()
-			break
+		fileName = of.GetName()
+		fileUri = of.GetUri()
+
+		if !strings.HasSuffix(fileName, outputsZipFileName) &&
+			!strings.HasSuffix(fileName, testXmlFileName) {
+			// Ignore this file.
+			continue
 		}
-	}
-	if len(outFileName) == 0 {
-		// The outputs.zip file was not found.
-		return nil
-	}
 
-	var zipFileCloser io.ReadCloser
-	var readErr error = nil
-	urlParts := strings.Split(outFileUri, "://")
-	if len(urlParts) != 2 {
-		return fmt.Errorf("Error reading %s file: malformed URL: %s", outFileName, outFileUri)
-	}
-	scheme, fileRef := urlParts[0], urlParts[1]
-	switch scheme {
-	case "bytestream":
-		// Handle cluster build scenario, translating bytestream:// URL to file URL for http.Get().
-		zipFileCloser, readErr = openBytestreamFile(outFileName, outFileUri)
-	case "file":
-		// Use URI without file:// prefix to access the local file system path.
-		zipFileCloser, readErr = openLocalFile(fileRef)
-	default:
-		// log and ignore this file: not a supported URL scheme prefix.
-		logger.Printf("Unexpected URI scheme when processing %s file: %s", outFileName, outFileUri)
-		return nil
-	}
-	if readErr != nil {
-		// Attempt to read the zip file failed.
-		return fmt.Errorf("Error reading %s file: %w", outFileName, readErr)
-	}
-	defer zipFileCloser.Close()
+		fileCloser, err := openOutputFile(fileName, fileUri)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("%w", err))
+			continue
+		}
+		defer fileCloser.Close()
 
-	// Process test metrics output file(s).
-	// Each output file contains a single (potentially large) protobuf message.
-	// For now, it's up to the client to split large metric datasets into multiple .metrics.pb files.
-	if err := processZip(stream, zipFileCloser); err != nil {
-		return fmt.Errorf("Error processing %s file: %w", outFileName, err)
+		// Found an output file to process. It is one whose file suffix
```

Comment:
I think the above would be better off as a switch or if/else ladder like:

```
switch {
case strings.HasSuffix(fname, outputsZipFileName):
  // open file
  // pass to processZipMetrics
case strings.HasSuffix(fname, testXmlSuffix):
  // open file
  // pass to process xml
default:
  continue
}
```

... so that it's more clear where to drop in additional handling for more file types

Reply:
OK, changed code to a single switch statement.